internal/compiler-bridge/src/main/scala/xsbt/API.scala

```diff
@@ -33,26 +34,49 @@ final class API(val global: CallbackGlobal) extends Compat with GlobalHelpers {
 
     def apply(unit: global.CompilationUnit): Unit = processUnit(unit)
 
-    def processUnit(unit: CompilationUnit) = if (!unit.isJava) processScalaUnit(unit)
-    def processScalaUnit(unit: CompilationUnit): Unit = {
+    private def processUnit(unit: CompilationUnit) = if (!unit.isJava) processScalaUnit(unit)
+
+    private def debugOutput(map: JavaMap[String, Array[String]]): String = {
+      val stringBuffer = new StringBuffer()
+      val it = map.entrySet().iterator()
```

Comment:
I think we can add commented scala code to document what is going on here (and similar loops). @jvican @eed3si9n @dwijnand WDYT?